### PR TITLE
Xpetra: cleanup and tests for blocked operators

### PR DESCRIPTION
**Pull request by mayrmt**
@trilinos/xpetra 

## Motivation

- add unit tests
- rename and document some variables

## Testing

Existing and new tests pass.

### CODE REVIEW

**Comment by mayrmt** [comment]
Pipeline fails due to random failures in Tempus (not related to this PR). See also the discussion in #3276, especially [this comment](https://github.com/trilinos/Trilinos/issues/3276#issuecomment-718256892) and below.

**Comment by mayrmt** [comment]
@trilinos/xpetra Autotester finally passed. I'm looking forward to your review and comments!

**Comment by william76** [comment]
@mayrmt  It looks like one of the nodes on the jenkins build farm (ascic166) has some problem which was causing the gcc 8.3.0 build to fail the four Tempus jobs. It was _only_ failing on that one node so I disabled that node from the build pool over the weekend so we can figure out what was wrong with it :p  If you see any more of those gcc 8.3.0 jobs failing on those Tempus jobs in the next couple of weeks let me know ;)

File: packages/xpetra/src/BlockedCrsMatrix/Xpetra_MapExtractorFactory.hpp [hunk 1]
```diff
@@ -64,20 +64,32 @@ namespace Xpetra {
...
-    MapExtractorFactory() {}
+    MapExtractorFactory() = delete;
 
   public:
-    /// \brief Constructor specifying the Maps.
-    ///
-    /// The Maps indirectly specify the linear algebra library to use
-    /// (Tpetra or Epetra).
+    /*!
+      @brief Build MapExtrasctor from a given set of partial maps.
```
Comment (cgcgcg) [inline packages/xpetra/src/BlockedCrsMatrix/Xpetra_MapExtractorFactory.hpp:71]: spelling

File: packages/xpetra/src/BlockedCrsMatrix/Xpetra_MapExtractorFactory.hpp [hunk 2]
```diff
@@ -64,20 +64,32 @@ namespace Xpetra {
...
     static Teuchos::RCP<Xpetra::MapExtractor<Scalar,LocalOrdinal,GlobalOrdinal,Node> >
     Build (const Teuchos::RCP<const Map>& fullmap,
            const std::vector<Teuchos::RCP<const Map> >& maps,
            bool bThyraMode = false)
     {
       return rcp (new Xpetra::MapExtractor<Scalar,LocalOrdinal,GlobalOrdinal,Node> (fullmap, maps, bThyraMode));
     }
+
+    /*!
+      @brief Build MapExtrasctor from a given BlockedMap.
```
Comment (cgcgcg) [inline packages/xpetra/src/BlockedCrsMatrix/Xpetra_MapExtractorFactory.hpp:84]: ditto

**Review by cgcgcg — APPROVED** [review]
Up to you, @mayrmt, if you want to fix the spelling now or later.

File: packages/xpetra/src/BlockedCrsMatrix/Xpetra_MapExtractorFactory.hpp (hunk 1 above)
Comment (mayrmt) [reply packages/xpetra/src/BlockedCrsMatrix/Xpetra_MapExtractorFactory.hpp:71]: Will push fix later.

File: packages/xpetra/src/BlockedCrsMatrix/Xpetra_MapExtractorFactory.hpp (hunk 2 above)
Comment (mayrmt) [reply packages/xpetra/src/BlockedCrsMatrix/Xpetra_MapExtractorFactory.hpp:84]: Will push fix later.

**Pull request merged**